mlir/lib/Dialect/Rock/Transforms/SugarToLoops.cpp

```diff
@@ -853,223 +857,333 @@ static Value computeMemRefNumElements(OpBuilder &b, Location loc,
-    {
-      OpBuilder thenb = ifb.getThenBodyBuilder(b.getListener());
-      // If the current value is positive doing a signed max integer comparison
-      // will be equivalent to a floating point max comparison.
-      thenb.create<amdgpu::RawBufferAtomicSmaxOp>(
-          loc, stValIntCasted, destMemRef, coordsI32,
-          /*boundsCheck=*/needHWBoundsCheck,
-          /*indexOffset=*/indexOffset,
-          /*sgprOffset=*/nullptr);
-    }
-    {
-      OpBuilder elseb = ifb.getElseBodyBuilder(b.getListener());
-      // If the current value is negative doing a unsigned min integer
-      // comparison will be equivalent to a floating point max comparison
-      // because signed bit will always make negative number larger. Moreover,
-      // if both are negative, then smaller of those unsigend integers should be
-      // the larger floating point value.
-      elseb.create<amdgpu::RawBufferAtomicUminOp>(
-          loc, stValIntCasted, destMemRef, coordsI32,
-          /*boundsCheck=*/needHWBoundsCheck,
-          /*indexOffset=*/indexOffset,
-          /*sgprOffset=*/nullptr);
-    }
-    return success();
-  }
 
-  LogicalResult
-  createAtomicFMax(BufferStoreOp op, Value stVal, Value destMemRef,
-                   SmallVectorImpl<Value> &coordsI32, bool needHWBoundsCheck,
-                   IntegerAttr indexOffset, PatternRewriter &b) const {
-    Location loc = op.getLoc();
-    bool hasAtomicFmaxF32 =
-        bitEnumContainsAll(op.getFeatures(), GemmFeatures::atomic_fmax_f32);
-    if (hasAtomicFmaxF32) {
-      b.create<amdgpu::RawBufferAtomicFmaxOp>(loc, stVal, destMemRef, coordsI32,
-                                              /*boundsCheck=*/needHWBoundsCheck,
-                                              /*indexOffset=*/indexOffset,
-                                              /*sgprOffset=*/nullptr);
-    } else {
-      LogicalResult result = createAtomicFMaxUsingIntegerAtomics(
-          op, stVal, destMemRef, coordsI32, needHWBoundsCheck, indexOffset, b);
-      if (failed(result)) {
-        return result;
-      }
-    }
-    return success();
+// Implement atomic floating point max.
+//
+// This _used_ to implement a clever hack for gfx9xx that used a type pun
+// to integer and then did an atomic signed_max/unsigned_min for
+// positive/negative floats, but, with large tensors, we can't do that
+// because upstream doesn't have the ability to do type puns on a memref.
+// We don't know if the smax/umin trick is faster than a CAS loop,
+// so we're keeping it around in the comments below for performance evaluation
+// once we have a reduction-utilizing client.
+//
+// Returns the operation that the underlying global_store could be replaced
+// with.
+static Operation *makeAtomicFmax(PatternRewriter &b, Location loc, Value data,
+                                 Value dest, ArrayRef<Value> coords,
+                                 bool useBuffers, bool useBufferOobChecks) {
```

Comment:
nit naming : it'd better to use useBufferIntrinsics as just buffers are too generic

Reply:
Updating to `useBufferOps` to match the rest of the code